src/frontend/src/handler/create_mv.rs

```diff
@@ -117,7 +108,17 @@ pub async fn handle_create_mv(
     let session = context.session_ctx.clone();
 
     let (table, graph) = {
-        let (plan, table) = gen_create_mv_plan(&session, context.into(), query, name, false)?;
+        {
+            let catalog_reader = session.env().catalog_reader().read_guard();
```

Comment:
An additional `read_guard` is introduced, which makes me still prefer `get` and `check` at the same place. How do you think?

Reply:
> Also, does (database, schema) uniquely identify a relation? The function returns Result<(DatabaseId, SchemaId)>, but you said we should name it get_relation_name.

Oops, I finally understand what you mean previously..

I think we can lookup `db` and `schema` outside and pass them to `check_relation_name_duplicated`.

Reply:
Seems like @skyzh  is ok with this. I will mark this as resolved.